snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/index/update/ReferenceSetMembershipUpdater.java

```diff
@@ -79,9 +79,7 @@ private void processReferencingRefSetIds(final Multiset<String> memberOf, final
 			.stream()
 			.filter(c -> c.getChangeKind() == MemberChangeKind.REMOVED)
 			.forEach(change -> {
-				if (change.isActive()) {
-					activeMemberOf.remove(change.getRefSetId());
-				}
```

Comment:
This `if` block is still required. 

`memberOf` and `activeMemberOf` also keep track of how many members in the same reference set refer to this component, so if there are eg. 5 members in R1 that have this component set as the referenced component, and 3 are active, the two multisets will have entries `R1: 5` and `R1: 3`, respectively.

- When an **addition** is reported, we always increase the counter for the reference set in `memberOf`, but only increase `activeMemberOf` if the new member is active;
- When a **change** in status<sup>1</sup> is reported, we check if the member was inactivated or re-activated, in which case we modify the counter in `activeMemberOf` accordingly, but do not touch `memberOf` at all;
- Finally, when a **removal** is reported, we always decrease the counter for `memberOf`, but should only do the same for `activeMemberOf` if the detached member was active!

<sup>1</sup> We should also check if callers register `MemberChangeKind.CHANGED` items for this reason only!

Reply:
The code related to footnote 1 works as expected: https://github.com/b2ihealthcare/snow-owl/blob/92045733c6033c377dd9a2a5fee12cca4eaf347a/snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/index/change/ReferringMemberChangeProcessor.java#L50-L58

Reply:
> This `if` block is still required.
> 
> `memberOf` and `activeMemberOf` also keep track of how many members in the same reference set refer to this component, so if there are eg. 5 members in R1 that have this component set as the referenced component, and 3 are active, the two multisets will have entries `R1: 5` and `R1: 3`, respectively.
> 
> * When an **addition** is reported, we always increase the counter for the reference set in `memberOf`, but only increase `activeMemberOf` if the new member is active;
> * When a **change** in status1 is reported, we check if the member was inactivated or re-activated, in which case we modify the counter in `activeMemberOf` accordingly, but do not touch `memberOf` at all;
> * Finally, when a **removal** is reported, we always decrease the counter for `memberOf`, but should only do the same for `activeMemberOf` if the detached member was active!
> 
> 1 We should also check if callers register `MemberChangeKind.CHANGED` items for this reason only!

Right, then @AAAlinaaa please revert this change.